### ISSUE TITLE
Make it possible to boot the full system from charging-sdl

### ISSUE THREAD

**Issue by ollieparanoid** [postmarketOS/charging-sdl#19]
Idea how to implement this:
* Add a button to the UI with the postmarketOS logo
* When the user touches it, we exit charging-sdl with a special code (e.g. `100`)
* In the initramfs code, we check for that exist code, and if it is the special one, we continue with normal booting.

**Comment by MartijnBraam** [comment]
Doesn't this just work if you call `reboot` from charging-sdl? That probably makes it boot outside of charging mode.

**Comment by ollieparanoid** [comment]
That would work, but it would be a lot slower (feels like 10 seconds or so on my devices from showing the OEM logo to osk-sdl).

Related: @drebrez found in https://github.com/postmarketOS/pmbootstrap/issues/1081#issuecomment-370619092 that at least one Android phone does it without the power cycle as well.

**Comment by MayeulC** [comment]
As long as the bootloader doesn't do funny stuff like disabling peripherals in charging mode, that could work. But I see no way of making sure it doesn't. In theory, that's possible ; has it ever been observed "in the wild"?
Well, it doesn't matter if Linux just reactivates them, but I feel like this is something we would have to keep in mind if strange stuff starts happening when booting in charging mode.

**Comment by ollieparanoid** [comment]
Good point! If we notice that it does not work with some devices, I would suggest a `deviceinfo` variable that forces the full reboot instead of the time-saving method described above.

**Comment by drebrez** [comment]
@ollieparanoid We should consider to boot to full system also when we long press the power button
